app/background.cc

```diff
@@ -0,0 +1,161 @@
+/* This is licensed software, @see LICENSE file.
+ * Authors - @see AUTHORS file. */
+
+#include <stdio.h>
+#include <time.h>
+#include <unistd.h>
+#include <thread>
+#include <mutex>
+#include <opencv2/videoio.hpp>
+#include <opencv2/imgproc.hpp>
+#include <opencv2/highgui.hpp>
+
+// Internal state of background processing
+struct background_t {
+    int dbg;
+    bool vid;
+    volatile bool run;
+    cv::VideoCapture cap;
+    int frm;
+    double fps;
+    cv::Mat raw;
+    std::thread thr;
+    std::mutex mux;
+};
+
+// Internal video reader thread
+static void read_thread(background_t *pbkd) {
+    if (pbkd->dbg) fprintf(stderr, "background: thread start\n");
+    struct timespec last;
+    struct timespec proc;
+    clock_gettime(CLOCK_REALTIME, &last);
```

Comment:
std::chrono -> monotonic_clock

Reply:
Yeah - I cut/past that from test code :smile: will fix.

Reply:
Especially note the difference between wall clock time and monotonic clock time …

Reply:
decided to use `steady_clock` (monotonic)